src/sqlfluff/rules/L046.py

```diff
@@ -38,21 +43,25 @@ class Rule_L046(BaseRule):
     targets_templated = True
 
     @staticmethod
-    def _get_whitespace_ends(s: str) -> Tuple[str, str, str]:
+    def _get_whitespace_ends(s: str) -> Tuple[str, str, str, str, str]:
```

Comment:
Could you add a doctest example for this routine? I.e. input string, then the values it returns?

https://docs.python.org/3/library/doctest.html

This will be useful mainly for documentation, but I _think_ pytest will automatically detect and execute it as well (much like a "normal" Python test).

Reply:
Great idea.

Reply:
Good idea. I've added doctests - but they weren't being picked up by tox, so I've added an explicit tox environment to run them. Not many in there yet but it's a useful environment for them.